### ISSUE TITLE
No need to place 'Embedded Objects' within Configuration <SchemaObject>[

### ISSUE THREAD

**Issue by dotjon0** [realm/realm-dart#1001]
### Description

'embedded objects' currently have to be included within the `Configuration <SchemaObject>[`. This feature request is to support not having to place 'embedded objects' within the `Configuration <SchemaObject>[`. This will increase maintainability and readability - a database with 70 models could easily have a few hundred 'embedded objects'...

Here is a related thread which provides an example #662 for context:

@RealmModel()
@MapTo('person')
class _Person {
  late String name;

  @MapTo('address')
  _Address? personAddress;
}

// The generated `Address` class will be an embedded object.
@RealmModel(ObjectType.embeddedObject)
class _Address {
  late String street;
  late String city;
}

```


### How important is this improvement for you?

Would be a major improvement

**Comment by nielsenko** [comment]
Actually, any dependency (embedded or not) could be automatically included, when adding a schema to a configuration, if we calculated the transitive dependencies in the generator.

**Comment by dotjon0** [comment]
any updates on this?

**Comment by nirinchev** [comment]
No, this is not on our immediate Todo list as it's a nice to have feature, but doesn't unlock any new use cases.

**Comment by dotjon0** [comment]
Hi @nielsenko thats for the update. For us, we have around 100 models, and around 300 embedded models.... So for us this is not a nice to have feature, its an essential to make this manageable...

**Comment by nielsenko** [comment]
Hi @dotjon0. Credit for update goes to @nirinchev .. I guess over github usernames are a bit too similar 😄

**Comment by nirinchev** [comment]
@dotjon0 fair enough - as is the case with most of these features - if it's something you deem critical for your project, definitely bring it up with your AE. They have mechanisms to influence the product roadmap and would be in best position to advise engineering on what features would have the broadest impact across our customers.
